frontend/javascripts/oxalis/geometries/materials/node_shader.ts

```diff
@@ -188,9 +188,9 @@ void main() {
       // the nodeId is encoded in the RGB channels as a 3 digit base-256 number in a number of steps:
       // - nodeId is divided by the first three powers of 256.
       // - each quotient is rounded down to the nearest integer (since the fractional part of each quotient is covered by a less significant digit)
-      // - each digit is divided by 256 again, since color values in OpenGL must be in the range [0, 1]
-      // - finally, the non-fractional part of each digit is removed (since it is covered by a more significant digit)
-      color = fract(floor(nodeId / vec3(256.0 * 256.0, 256.0, 1.0)) / 256.0);
+      // - each digit is mod'ed by 256 so that the relevant significance is encoded
+      // - finally, the number is divided by 255 because the color values in WebglGL must be in the range [0, 1]
+      color = mod(floor(nodeId / vec3(256.0 * 256.0, 256.0, 1.0)), 256.0) / 255.0;
```

Comment:
As the code comment outlines, this line (1) encodes the number using base 256 and (b) converts the numbers to a range of [0, 1]. The latter has to happen by dividing by 255.

The previous implementation used a division and `fract()` to do the same (`fract(x / 256.)` vs `mod(x, 256.) / 255.)`, but that couldn't account for the 256 mod as well as the 255 scaling.

I hope this makes sense! If not, let's have a call.